src/tasks.ts

```diff
@@ -75,29 +73,38 @@ export const withRetries = <E, T>(
         return currentTask;
       }
       // allow one run of the task
-      return currentTask.orElse(l => {
-        // if the task fails...
-        if (mustAbort) {
-          return fromLeft(RetryAborted);
-        }
-        if (l === TransientError) {
-          // ...with a TransientError, chain it with a backoff delay
-          // an then with another run.
-          const delay = new TaskEither<
-            E | TransientError | RetryAborted,
-            boolean
-          >(delayTask(backoff(count), true).map(d => right(d)));
-          return delay.chain(() => runTaskOnce(count + 1, currentTask));
-        }
-        // ...with an error that is not a TransientError, we just return it
-        return fromLeft(l);
-      });
+      return pipe(
+        currentTask,
+        TE.orElse(l => {
+          // if the task fails...
+          if (mustAbort) {
+            return TE.left(RetryAborted);
+          }
+          if (l === TransientError) {
+            // ...with a TransientError, chain it with a backoff delay
+            // an then with another run.
+            const delay = TE.of(
+              pipe(
+                delayTask(backoff(count), true),
+                T.map(d => right(d))
+              )
+            );
+            return pipe(
+              delay,
+              TE.chain(() => runTaskOnce(count + 1, currentTask))
+            );
```

Comment:
It seems to me that there's no need to lift a Task to a TaskEither, we can just write
```suggestion
            return pipe(
              delayTask(backoff(count), true),
              T.map(d => right(d)),
              TE.chain(() => runTaskOnce(count + 1, currentTask))
            );
```